src/lib/RoomConnection.ts

```diff
@@ -539,6 +558,35 @@ export default class RoomConnection extends TypedEventTarget {
         this.dispatchEvent(new CustomEvent("streaming_stopped"));
     }
 
+    private _handleScreenshareStarted(screenshare: SignalScreenshareStartedEvent) {
+        const { clientId: participantId, streamId, hasAudioTrack } = screenshare;
+        const remoteParticipant = this.remoteParticipants.find((p) => p.id === participantId);
+
+        if (!remoteParticipant) {
+            this.logger.log("WARN: Could not find participant for screenshare");
+            return;
+        }
+        remoteParticipant.addStream(streamId, "to_accept");
+        this._handleAcceptStreams([remoteParticipant]);
+        this.screenshares = [...this.screenshares, { participantId, streamId, hasAudioTrack, stream: undefined }];
```

Comment:
I noticed that the ongoing screenshare appears 2x on signal reconnect, and 3x on the next reconnect, etc.

This could row above could be the reason for it: the new room_join's (reconnect's) screenshare client is just appended to the existing screenshare list, even if the same participantId+streamId is already there.

I guess they should be pushed only if the streamId (or participantId?) is not there yet. (maybe it doesn't matter which one you use as 1 client can start 1 screenshare atm)

<img width="654" alt="Screenshot 2023-09-06 at 15 30 31" src="https://github.com/whereby/browser-sdk/assets/1159931/d8914329-6f2f-41ea-a2c4-6188357d8037">

You can test this by reloading your local signal server by saving something on the backend.

Reply:
Fixed. We still have issues with events triggering twice, but there's a separate ticket for that.